roxie/roxiemem/roxiemem.cpp

```diff
@@ -628,12 +637,12 @@ static void *suballoc_aligned(size32_t pages, bool returnNullWhenExhausted)
         unsigned i;
         for (i = heapLWM; i < heapBitmapSize; i++)
         {
-            unsigned hbi = heapBitmap[i];
+            heap_t hbi = heapBitmap[i];
             if (hbi)
             {
                 const unsigned pos = countTrailingUnsetBits(hbi);
-                const unsigned mask = 1U << pos;
-                const unsigned match = i*UNSIGNED_BITS + pos;
+                const heap_t mask = ((heap_t)1U) << pos;
```

Comment:
Sure.  I might have done something like:

ifdef _ARCH_PPC64EL_
# define ALL_BITS 1UL
#else
# define ALL_BITS 1U
#endif

And used ALL_BITS everywhere.  But I'm sure its promoted correctly.